packages/dendron-next-server/pages/workspace/config.tsx

```diff
@@ -0,0 +1,326 @@
+  FormHelperText,
+  FormLabel,
+  Heading,
+  Input,
+  Stack,
+  Switch,
+  Text,
+  Tooltip,
+} from "@chakra-ui/react";
+import { get } from "lodash";
+
+const getConfigData = (): { data: DendronConfig } => {
+  return {
+    data: {
+      vaults: [
+        {
+          fsPath: "vault",
+          name: "",
+        },
+      ],
+      site: {
+        assetsPrefix: "",
+        copyAssets: false,
+        siteHierarchies: ["dendron"],
+        siteNotesDir: "",
+        siteRepoDir: "",
+        siteRootDir: "docs",
+        usePrettyRefs: false,
+      },
+    },
+  };
+};
+
+// const saveConfigData = (data: DendronConfig) => {
+//   console.log("data saved");
+//   console.log(JSON.stringify(data, null, 2));
+// };
+
+type InputControlProps = {
+  name: string;
+  label?: string;
+  placeholder: string;
+  help: string | ReactNode;
+  disabled?: boolean;
+};
+
+function InputControl({
+  name,
+  label,
+  placeholder,
+  help,
+  disabled,
+}: InputControlProps) {
+  return (
+    <Field name={name}>
+      {({ form: { errors, touched } }) => (
+        <FormControl
+          isInvalid={
+            /**
+             * If field has errors AND they've interacted with this field, mark
+             * control as invalid. */
+            !!(get(errors, name) && get(touched, name))
+          }
+        >
+          {!!label && <FormLabel htmlFor={name}>{label}</FormLabel>}
+
+          <Field
+            //
+            as={Input}
+            name={name}
+            placeholder={placeholder}
+            id={name}
+            disabled={disabled}
+          />
+
+          <FormErrorMessage>{errors.site?.siteRootDir}</FormErrorMessage>
+
+          <FormHelperText>{help}</FormHelperText>
+        </FormControl>
+      )}
+    </Field>
+  );
+}
+
+export default function ConfigSamplePage() {
+  // const { data: configData } = getConfigData();
+
+  const [configData, saveConfigData] = useState(getConfigData().data);
+
+  React.useEffect(() => {
+    console.log("save", JSON.stringify(configData, null, 2));
+  }, [JSON.stringify(configData)]);
+
+  return (
+    <Container paddingBottom={12}>
+      <Stack spacing={4}>
+        <Heading>Config Data</Heading>
+
+        <Formik
+          enableReinitialize
+          initialValues={configData}
+          onSubmit={(formValues) => saveConfigData(formValues)}
+        >
+          {({
+            touched,
+            errors,
+            values,
+            resetForm, // NOTE only works on fields which have configData keys
+          }) => (
+            <Box as={Form} onChange={console.log}>
+              <Stack spacing={8}>
+                <Box as="fieldset">
+                  <Heading size="md" as="legend">
+                    Vaults
+                  </Heading>
+
+                  <Stack spacing={4}>
+                    {configData.vaults.map((vault, index) => (
+                      <Tooltip
+                        label="Vaults are not yet editable via this form."
```

Comment:
change to: "You can update vaults using [Dendron Vault Commands](https://dendron.so/notes/eea2b078-1acc-4071-a14e-18299fc28f47.html#vault-add)

Reply:
:heavy_check_mark: @ 2177b0f2